Framework/DataHandling/src/SaveTBL.cpp

```diff
@@ -83,16 +83,28 @@ void SaveTBL::exec() {
         std::string groupHeading = columnHeadings[columnIndex];
         if (ws->getColumn(groupHeading)->type() != "int") {
           file.close();
-          remove(filename.c_str());
-          throw std::runtime_error(groupHeading +
-                                   " Column must be of type \"int\"");
+          int error = remove(filename.c_str());
+          if (error == 0)
+            throw std::runtime_error(groupHeading + " column must be of type "
+                                                    "\"int\". The TBL file "
+                                                    "will not be saved.");
+          else
+            throw std::runtime_error(
```

Comment:
Can we add something in to the error comment about the file not being removed and it being incorrect? If it's not removed then someone may see it and think it is okay?